### DISCUSSION TITLE
Phoneme pronunciation speed (for syncing with phoneme text output)

### DISCUSSION THREAD

**Discussion by fbcosentino** [espeak-ng/espeak-ng#1780] [General]
Hi, I am trying to find out/understand how espeak-ng deals with pronunciation speed for phonemes.

I know I can output the phonemes with `-x` and I would like to create a script to timestamp them to match the audio output

I speak a bit of C so if instead of an actual answer you guys point me to the lines in the source which deal with that I can try to get my answer from there

**Comment by fbcosentino** [comment]
...or, if it's unpredictable, I can try to contribute to the code adding a feature to output timestamped phonemes instead of just phonemes, and make a PR

but in that case I might need some guidance (a short paragraph) on how/where this is calculated

**Comment by jaacoppi** [comment]
I doubt that this feature is useful for many people so might not be worth a PR.

You can start digging with:
1) run  espeak-ng -s <WPM> and see how it's handled in code
2) search for "speed" or "wpm" in the source files. I warn you, there be dragons.

**Reply by fbcosentino** [reply]
Thanks!
This feature would allow espeak output to be used for mouth animation in sync with audio, which could serve both traditional pre-rendered character animation as well as animated virtual assistants, so it's not that useless. 
I'll try the dragons
